code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -0,0 +1,335 @@
+#define SOLID 1
+#define LIQUID 2
+#define GAS 3
+
+#define REM REAGENTS_EFFECT_MULTIPLIER
+
+datum/reagent/silver_sulfadiazine
+	name = "Silver Sulfadiazine"
+	id = "silver_sulfadiazine"
+	description = "100% chance per cycle of healing 2 points of BURN damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 2
+
+datum/reagent/silver_sulfadiazine/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.adjustFireLoss((volume-(volume*2))*REM)
+		M << "<span class='notice'>You feel your burns healing!</span>"
+		M.emote("scream")
+	if(method == INGEST)
+		M.adjustToxLoss(0.5*volume)
+		M << "<span class='notice'>You probably shouldn't have eaten that. Maybe you should of splashed it on, or applied a patch?</span>"
+	..()
+	return
+
```

Comment:
how that equation works:

let's say we have a volume of 50

it's 50-(50*2)

since adjusting by a negative value heals, it will heal 50